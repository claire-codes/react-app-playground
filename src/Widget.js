import React from "react";
import styled from "styled-components";

const StyledWidget = styled.div`
  height: 50px;
  width: 200px;
  background-color: ${props => (props.invisible ? "transparent" : "#61dafb")};
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
  ${props =>
    props.invisible &&
    `box-shadow: grey 5px 5px;
        margin-bottom: 10px;`};
`;

const Widget = ({ children, invisible }) => (
  <StyledWidget invisible={invisible}>{children}</StyledWidget>
);

export default Widget;
