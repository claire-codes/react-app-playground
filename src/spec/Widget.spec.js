import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import Widget from "../Widget";

test("it works", () => {
  const tree = renderer.create(<Widget />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule("display", "flex");
});
