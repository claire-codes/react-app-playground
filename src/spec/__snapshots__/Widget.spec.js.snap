// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c0 {
  height: 50px;
  width: 200px;
  background-color: #61dafb;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
/>
`;
